### PR TITLE
Ability to restrict cancelation of reservations upto x hours prior to the start of event

### PR DESCRIPTION
**Pull request by shlok007**

### CODE REVIEW

**Comment by shlok007** [comment]
Solution to #48 
Kindly review it.

File: db/schema.rb [hunk 1]
```diff
@@ -50,6 +47,7 @@
     t.boolean  "repeatable"
     t.datetime "created_at",                          null: false
     t.datetime "updated_at",                          null: false
+    t.integer  "modifiable_before",      default: 0
```
Comment (shubhpatel108) [inline db/schema.rb]: Might have to think about the default value. Does 0 makes it unmodifiable or modifiable at any time?
If it is the former, @jessyurko shall we keep the x = 48 hours (issue: #48)?

Comment (shlok007) [inline db/schema.rb]: It will make it modifiable at any time before the start of the event.

Comment (jessyurko) [inline db/schema.rb]: 48 sounds good to me

File: spec/models/reservation_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe Reservation, type: :model do 
+	before(:each) do
+      @reservation = create(:reservation)
```
Comment (shubhpatel108) [inline spec/models/reservation_spec.rb]: Not well-indented. Also, Use 2 spaces, no tabs.

Comment (shubhpatel108) [inline spec/models/reservation_spec.rb]: The text editor will have options to interconvert tabs and spaces (as well as their sizes).

File: app/models/reservation.rb [hunk 3]
```diff
@@ -73,8 +73,12 @@ def confirmed?
     status == Status::CONFIRMED
   end
 
+  def cancelable?
+    (Status::CANCELABLE.include? status) && ((self.start.to_i-Time.now.to_i) > (self.nook.modifiable_before*3600))
+  end
+
   def modifiable?
-    Status::MODIFIABLE.include? status
+    (Status::MODIFIABLE.include? status) && ((self.start.to_i-Time.now.to_i) > (self.nook.modifiable_before*3600))
```
Comment (shubhpatel108) [inline app/models/reservation.rb:81]: @shlok007 Sorry for noticing this late - though you have updated this function, it was in fact used for admin. The 'My reservations' for patron view still give the option to modify.

Comment (shlok007) [inline app/models/reservation.rb:81]: @shubhpatel108 From what I understand #143 should fix this too.

Comment (shubhpatel108) [inline app/models/reservation.rb:81]: @shlok007 It seems that the 'modify' button will still show up.

Comment (shlok007) [inline app/models/reservation.rb:81]: @shubhpatel108 To fix that a simple condition to restrict the button in the views should suffice, I'll push another commit fixing this in #143 .

Comment (shubhpatel108) [inline app/models/reservation.rb:81]: @shlok007 Thanks :)

**Pull request merged**